hs/src/Reach/Eval/Core.hs

```diff
@@ -492,31 +492,75 @@ dlvToDL = \case
   DLV_Data _ t s mdv -> DLAE_Data t s <$> rec mdv
   _ -> Nothing
   where
-    rec  = maybe Nothing dlvToDL
+    rec  = dlvToDL
     recAssoc (k, v) = (,) k <$> rec v
 
-slToDLExportVal :: SLVal -> App (Maybe DLExportVal)
-slToDLExportVal v = slToDLV v <&> maybe Nothing dlvToEV
+slToDLExportVal :: SLVal -> App (Maybe DLExportBlock)
+slToDLExportVal v = slToDLV v >>= maybe (return Nothing) (dlvToEV >=> (return . Just))
 
-dlvToEV :: DLValue -> Maybe DLExportVal
+dlvToEV :: DLValue -> App DLExportBlock
 dlvToEV = \case
-  DLV_Arg at a  -> return $ DLEV_Arg at a
-  DLV_Fun at args b -> return $ DLEV_Fun at args b
-  DLV_Array at dt mdvs -> return $ DLEV_LArg at $ DLLA_Array dt (recs mdvs)
-  DLV_Tuple at mdvs -> return $ DLEV_LArg at $ DLLA_Tuple (recs mdvs)
-  DLV_Obj at menv ->
-    return $ DLEV_LArg at $ DLLA_Obj $ M.fromList $ mapMaybe recAssoc $ M.toList menv
-  DLV_Data at t s mdv -> DLEV_LArg at . DLLA_Data t s <$> rec mdv
-  DLV_Struct at mdvs -> return $ DLEV_LArg at $ DLLA_Struct $ mapMaybe recAssoc mdvs
+  DLV_Arg at a -> return $ DLExportBlock Seq.empty $ DLEV_Arg at a
+  DLV_Fun at a b -> return $ DLExportBlock Seq.empty $ DLEV_Fun at a b
+  DLV_Array at dt mdvs -> locAt at $ do
```

Comment:
This function now seems extremely similar to the `dlvToDL` composed with `compileDLArgExpr` (or whatever its called)